### PR TITLE
Enable different upper and lower BCs for Brick.

### PR DESCRIPTION
**Pull request by jyoo1042**
## Proposed changes
Enable different upper and lower boundary conditions in Brick.

### CODE REVIEW

File: tests/Unit/Domain/Creators/Test_Brick.cpp [hunk 1]
```diff
@@ -117,15 +117,20 @@ void test_brick() {
                                {Direction<3>::upper_eta()},
                                {Direction<3>::lower_zeta()},
                                {Direction<3>::upper_zeta()}}});
+  using var_wrap = std::variant<
```
Comment (kidder) [inline tests/Unit/Domain/Creators/Test_Brick.cpp]: rename var_wrap to VariantType

File: tests/Unit/Elliptic/DiscontinuousGalerkin/SubdomainOperator/Test_SubdomainOperator.cpp [hunk 2]
```diff
@@ -675,6 +675,12 @@ struct InitializeConstitutiveRelation {
 // handles all of these geometries correctly.
 // [[TimeOut, 25]]
 SPECTRE_TEST_CASE("Unit.Elliptic.DG.SubdomainOperator", "[Unit][Elliptic]") {
+  // Needed for Brick
+  using var_wrap = std::variant<
```
Comment (kidder) [inline tests/Unit/Elliptic/DiscontinuousGalerkin/SubdomainOperator/Test_SubdomainOperator.cpp]: var_wrap -> VariantType

File: tests/Unit/Elliptic/DiscontinuousGalerkin/Test_DgOperator.cpp [hunk 3]
```diff
@@ -409,6 +409,12 @@ void test_dg_operator(
 }  // namespace
 
 SPECTRE_TEST_CASE("Unit.Elliptic.DG.Operator", "[Unit][Elliptic]") {
+  // Needed for Brick
+  using var_wrap = std::variant<
```
Comment (kidder) [inline tests/Unit/Elliptic/DiscontinuousGalerkin/Test_DgOperator.cpp]: var_wrap -> VariantType

File: tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Test_BoundaryConditionGhostData.cpp [hunk 4]
```diff
@@ -93,11 +93,16 @@ void test(const BoundaryConditionType& boundary_condition) {
   const std::array<size_t, 3> refinement_levels{0, 0, 0};
   const std::array<size_t, 3> number_of_grid_points{num_dg_pts, num_dg_pts,
                                                     num_dg_pts};
+  using var_wrap = std::variant<
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Test_BoundaryConditionGhostData.cpp]: var_wrap -> VariantType

File: tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/Test_BoundaryConditionGhostData.cpp [hunk 5]
```diff
@@ -88,11 +88,16 @@ void test(const BoundaryConditionType& boundary_condition,
   const std::array<size_t, 3> refinement_levels{0, 0, 0};
   const std::array<size_t, 3> number_of_grid_points{num_dg_pts, num_dg_pts,
                                                     num_dg_pts};
+  using var_wrap = std::variant<
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/Test_BoundaryConditionGhostData.cpp]: var_wrap -> VariantType

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by yoonso0-0** [comment]
This looks like a very nice feature to add.

**Review by kidder — APPROVED** [review]

**Pull request merged**